src/moscot/backends/ott/output.py

```diff
@@ -238,6 +241,227 @@ def _ones(self, n: int) -> ArrayLike:  # noqa: D102
         return jnp.ones((n,))
 
 
+class OTTNeuralOutput(BaseNeuralOutput):
+    """Output wrapper for GENOT."""
+
+    def __init__(self, model: GENOT, logs: dict[str, list[float]]):
+        """Initialize `OTTNeuralOutput`.
+
+        Parameters
+        ----------
+        model : GENOT
+            The OTT-Jax GENOT model
+        """
+        self._logs = logs
+        self._model = model
+
+    @property
+    def logs(self):
+        """Logs of the training.  A dictionary containing what the numeric values are i.e., loss.
+
+        Returns
+        -------
+        dict[str, list[float]]
+        """
+        return self._logs
+
+    def _project_transport_matrix(
+        self,
+        src_dist: ArrayLike,
+        tgt_dist: ArrayLike,
+        forward: bool,
+        func: Callable[[jnp.ndarray], jnp.ndarray],
+        save_transport_matrix: bool = False,  # TODO(@MUCDK) adapt order of arguments
+        batch_size: int = 1024,
+        k: int = 30,
+        length_scale: Optional[float] = None,
+        seed: int = 42,
+    ) -> sp.csr_matrix:
+        row_indices: Union[jnp.ndarray, List[jnp.ndarray]] = []
+        column_indices: Union[jnp.ndarray, List[jnp.ndarray]] = []
+        distances_list: Union[jnp.ndarray, List[jnp.ndarray]] = []
+        if length_scale is None:
+            key = jax.random.PRNGKey(seed)
+            src_batch = src_dist[jax.random.choice(key, src_dist.shape[0], shape=((batch_size,)))]
+            tgt_batch = tgt_dist[jax.random.choice(key, tgt_dist.shape[0], shape=((batch_size,)))]
+            length_scale = jnp.std(jnp.concatenate((func(src_batch), tgt_batch)))
+        for index in range(0, len(src_dist), batch_size):
+            distances, indices = get_nearest_neighbors(func(src_dist[index : index + batch_size, :]), tgt_dist, k)
+            distances = jnp.exp(-((distances / length_scale) ** 2))
+            distances /= jnp.expand_dims(jnp.sum(distances, axis=1), axis=1)
+            distances_list.append(distances.flatten())
+            column_indices.append(indices.flatten())
+            row_indices.append(
+                jnp.repeat(jnp.arange(index, index + min(batch_size, len(src_dist) - index)), min(k, len(tgt_dist)))
+            )
+        distances = jnp.concatenate(distances_list)
+        row_indices = jnp.concatenate(row_indices)
+        column_indices = jnp.concatenate(column_indices)
+        tm = sp.csr_matrix((distances, (row_indices, column_indices)), shape=[len(src_dist), len(tgt_dist)])
+        if forward:
+            if save_transport_matrix:
+                self._transport_matrix = tm
+        else:
+            tm = tm.T
+            if save_transport_matrix:
+                self._inverse_transport_matrix = tm
+        return tm
+
+    def project_transport_matrix(  # type:ignore[override]
```

Comment:
I know this is my bad as you inherited. But can we rename this to `project_to_transport_matrix`